### ISSUE TITLE
Stuttering response on Agent

### ISSUE THREAD

**Issue by Festiu** [UltimateHackingKeyboard/agent#2146]
As title says, I find like some stroboscopic response while using Agent.

I have recorded a video that is self-explanatory.
[https://streamable.com/aijct5](https://streamable.com/aijct5)

I don't know what is the reason, but these are some things to keep in mind
- This is a brand new computer (AM5 architecture with Ryzen 5 7600 CPU)
- New formatted OS (Win10 updated to june 2023)
- I've installed old versions of the Agent (2.x), and it has the stuttering too
- The Agent it's working ok on all levels (save profiles, flash firmware... all ok), it's just it's anoying to use
- When the Agent is on, the keyboard flashes slightly at the same rate as the Agent flashes (one flash every 2 seconds approx.). But it's a very very very subtle flicker, it's barely noticeable if you don't look closely.
- On my old computer there is no stuttering and works ok (2011 Intel Sandy Bridge)
- The Agent has stuttering too even with the UHK disconected

**Comment by mondalaci** [comment]
This issue must be specific to your computer since no one has ever reported something like this.

1. Does the process manager show any CPU-intensive tasks?
2. Can you reproduce this issue with other applications, preferably Electron-based applications?
3. Can you reproduce this issue on other computers?

**Comment by Festiu** [comment]
Hi there 

1. The CPU is fine, idling at 3% max. The Agent don't cause any load.
2. Good thing to know! I've just installed Discord for Windows and TelegramDesktop, and working fine.
3. Yes, this is homework for me. I'll post further results when I have some clue on this regard.

I know it's an uncomfortable position for you guys, since no one has reported a similar problem and it's hard to know what to do.

Well, I knew it wouldn't be easy, but at least I feel relieved to have created the issue. Who knows if hopefully we find the solution, or at least serve for other users in the future, even if it is to comment on the issue, you never know.

It remains for me to do tests with other computers, using the same OS (same .iso) and my same configuration, etc. I will post again when something clearer.

Thanks for the assistance.

**Comment by mondalaci** [comment]
Yes, please keep us posted. Maybe somebody will eventually have a similar issue, and then your post will turn out to be very helpful.

**Comment by mondalaci** [comment]
I'm closing this because it's a one-off, but please keep us posted; if we identify a bug, I'll reopen it.

**Comment by Festiu** [comment]
My latest tests are as follows:

- On a freshly installed Win10, the problem persists. I used the most updated official Win10 .iso with no tweaks.

- If I disconnect all the USB peripherals except for UHK, the stuttering is much less noticeable (but still present). However, as soon as I connect a mouse, the stuttering becomes heavy (I've tried 3 different models). Even with a mouse connected and UHK disconnected, the stuttering is pronounced. Other USB peripherals may interfere too, but the mouse is the most noticeable.

- I've explored all the BIOS options and updated the motherboard's firmware (MSI Pro B650-S Wifi), but nothing has changed.

- I've installed Debian 12, and it's working great there using the appimage, with no stuttering of any kind.

My conclusion is that there's something to do with the motherboard's USB ports. It's a 2023 model, and I wouldn't be surprised if a future firmware update solves the problem or if Windows updates itself. I haven't tested it with Win11 though.

Time will tell if some update brings about any changes.
